factory-reset-external

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+#
+# External storage factory reset for Sailfish OS.
+#
+# Copyright (C) 2017 Jolla Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2
+# of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# Parameters:
+# $@ -> Device paths of partitions to reformat.
+
+format_script()
+{
+cat << 'EOF'
+#!/bin/sh
+
+function log
+{
+        echo "$0: $1"
+}
+
+for device in $@; do
+	unset TYPE
+	unset LABEL
+	unset UUID
+
+	if [ ! -e "$device" ]; then
+		continue
+	fi
+
+        eval "$(/sbin/blkid -c /dev/null -o export $device)"
+
+	case "$TYPE" in
+		vfat)
+			format_command=/sbin/mkfs.vfat
+			format_arguments=()
+			if [ ! -z "$UUID" ]; then
+				format_arguments=( "${format_arguments[@]}" -i "${UUID//-}" )
+			fi
+			if [ ! -z "$LABEL" ]; then
+				format_arguments=( "${format_arguments[@]}" -n "${LABEL}" )
+			fi
+			;;
+		ext4)
+			format_command=/sbin/mkfs.ext4
+			format_arguments=( -F -E root_owner=100000:100000 )
+			if [ ! -z "$UUID" ]; then
+				format_arguments=( "${format_arguments[@]}" -U "${UUID}" )
+			fi
+                        if [ ! -z "$LABEL" ]; then
+				format_arguments=( "${format_arguments[@]}" -L "${LABEL}")
+			fi
+			;;
+		*)
+			if [ -z "$TYPE" ]; then
+				continue
+			fi
+
+			format_command=
+			format_arguments=
+			;;
+	esac
+
+	if [ -f $format_command ]; then
+                log "Formatting $device as $TYPE"
+
+                if test "$SAILFISHOS_WIPE_PARTITIONS" = "1"; then
+                        dd if=/dev/zero of=$device bs=1M
+                fi
+
+                echo $format_command "${format_arguments[@]}" $device
+                $format_command "${format_arguments[@]}" $device || log "Failed to format $device"
+	elif [ ! -z "$format_command" ]; then
+		TEMPMOUNT=$(mktemp -d)
```

Comment:
Adding comment here that states something about: "If filesystem format tool is not available, we remove all files and write disk full of zero to clean the volume."